File: scripts/optimize-images.js
const fs = require('fs');
const path = require('path');
const sharp = require('sharp');

const imageDir = path.join(__dirname, '../public/zz');
const backupDir = path.join(imageDir, 'backup');

// 백업 폴더 생성
if (!fs.existsSync(backupDir)) {
  fs.mkdirSync(backupDir, { recursive: true });
}

// 이미지 최적화 함수
async function optimizeImage(inputPath, outputPath) {
  try {
    // 원본 백업
    const backupPath = path.join(backupDir, path.basename(inputPath));
    if (!fs.existsSync(backupPath)) {
      fs.copyFileSync(inputPath, backupPath);
    }

    const metadata = await sharp(inputPath).metadata();
    const originalSize = fs.statSync(inputPath).size;

    let pipeline = sharp(inputPath);

    // 큰 이미지는 리사이즈
    const maxSize = 1920;
    if (metadata.width > maxSize || metadata.height > maxSize) {
      pipeline = pipeline.resize(maxSize, maxSize, {
        fit: 'inside',
        withoutEnlargement: true
      });
    }

    // PNG를 JPG로 변환하고 최적화
    if (path.extname(inputPath).toLowerCase() === '.png') {
      await pipeline
        .jpeg({ quality: 85, progressive: true })
        .toFile(outputPath);
    } else {
      // JPG 재압축
      await pipeline
        .jpeg({ quality: 85, progressive: true })
        .toFile(outputPath);
    }

    const newSize = fs.statSync(outputPath).size;
    const reduction = ((1 - newSize / originalSize) * 100).toFixed(1);
    
    console.log(`${path.basename(inputPath)}: ${(originalSize/1024).toFixed(1)}KB -> ${(newSize/1024).toFixed(1)}KB (${reduction}% 감소)`);
    
  } catch (error) {
    console.error(`Error processing ${inputPath}:`, error);
  }
}

// 이미지 처리
async function processImages() {
  const files = fs.readdirSync(imageDir);
  
  for (const file of files) {
    if (file.match(/\.(png|jpg|jpeg)$/i) && file !== 'backup') {
      const inputPath = path.join(imageDir, file);
      let outputPath;
      
      if (file.endsWith('.png')) {
        outputPath = path.join(imageDir, file.replace('.png', '_optimized.jpg'));
      } else {
        outputPath = path.join(imageDir, file.replace(/\.(jpg|jpeg)$/i, '_optimized.jpg'));
      }
      
      await optimizeImage(inputPath, outputPath);
    }
  }
  
  console.log('\n이미지 최적화 완료!');
  console.log('원본 파일은 backup 폴더에 저장되었습니다.');
}

processImages().catch(console.error);

File: optimize-images.py
import os
from PIL import Image
import shutil

# 이미지 경로
image_dir = '/home/tesola/ttss/tesolafixjs/public/zz'
backup_dir = os.path.join(image_dir, 'backup')

# 백업 폴더 생성
if not os.path.exists(backup_dir):
    os.makedirs(backup_dir)

# 이미지 압축 함수
def compress_image(input_path, output_path, quality=85):
    try:
        # 원본 백업
        backup_path = os.path.join(backup_dir, os.path.basename(input_path))
        if not os.path.exists(backup_path):
            shutil.copy2(input_path, backup_path)
        
        # 이미지 열기
        with Image.open(input_path) as img:
            # RGBA를 RGB로 변환 (JPG는 알파 채널 지원 안함)
            if img.mode == 'RGBA':
                rgb_img = Image.new('RGB', img.size, (255, 255, 255))
                rgb_img.paste(img, mask=img.split()[3])
                img = rgb_img
            
            # 원본 크기 확인
            width, height = img.size
            
            # 너무 큰 이미지는 리사이즈
            max_size = 1920
            if width > max_size or height > max_size:
                ratio = min(max_size/width, max_size/height)
                new_width = int(width * ratio)
                new_height = int(height * ratio)
                img = img.resize((new_width, new_height), Image.Resampling.LANCZOS)
            
            # 저장
            img.save(output_path, 'JPEG', quality=quality, optimize=True)
            
            # 파일 크기 비교
            original_size = os.path.getsize(input_path)
            new_size = os.path.getsize(output_path)
            
            print(f"{os.path.basename(input_path)}: {original_size/1024:.1f}KB -> {new_size/1024:.1f}KB ({(1-new_size/original_size)*100:.1f}% 감소)")
            
    except Exception as e:
        print(f"Error processing {input_path}: {e}")

# PNG와 JPG 파일 찾기
for filename in os.listdir(image_dir):
    if filename.endswith(('.png', '.jpg', '.jpeg')):
        input_path = os.path.join(image_dir, filename)
        
        # PNG는 JPG로 변환
        if filename.endswith('.png'):
            output_path = os.path.join(image_dir, filename.replace('.png', '_optimized.jpg'))
        else:
            output_path = os.path.join(image_dir, filename.replace('.jpg', '_optimized.jpg').replace('.jpeg', '_optimized.jpg'))
        
        compress_image(input_path, output_path)

print("\n이미지 최적화 완료!")
print("원본 파일은 backup 폴더에 저장되었습니다.")